estate/models/estate_property.py

```diff
@@ -32,4 +32,56 @@ class EstateProperty(models.Model):
             ('sold','Sold'),
             ('cancelled','Cancelled'),
         ], string="State", default='new', required=True, copy=False)
-    property_type_id = fields.Many2one("res.partner", string="Property Type")
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    seller_id = fields.Many2one("res.users", string="Seller", index=True, tracking=True, default=lambda self: self.env.user)
+    tag_ids = fields.Many2many("estate.property.tag", string="Property Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer(compute="_compute_total_area", string="Total area")
+    best_price = fields.Float(compute="_compute_best_price", string="Best price")
+
+    _sql_constraints = [
+        ('check_price', 'CHECK(expected_price > 0.0)', 'The expected price must be strictly positive.'),
+        ('check_selling_price', 'CHECK(selling_price > 0.0)', 'The Selling price must be strictly positive'),
+    ]
+    
+    @api.depends("living_area","garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            offer_ids = record.offer_ids
+            if offer_ids:
+                record.best_price = max(offer_ids.mapped('price'))
+            else:
+                record.best_price = 0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = ''
+
+    @api.constrains("selling_price","expected_price")
```

Comment:
```suggestion
    @api.constrains("selling_price", "expected_price")
```
Please take care of spaces wherever necessary. Follow coding guidelines.

Reply:
Noted!